test/test_nn.py

```diff
@@ -16387,6 +16389,50 @@ def cosine_distance(x, y):
             self.assertEqual(functional, modular, atol=1e-6, rtol=1e-6)
             self.assertEqual(traced, modular, atol=1e-6, rtol=1e-6)
 
+    @onlyOnCPUAndCUDA
+    def test_cross_entropy_loss_with_soft_labels(self, device):
```

Comment:
nit:  It would be nice to test correctness for some cases where `target` is not one-hot-encoded

Reply:
Note that the tests generated from the `criterion_tests` entry check against a reference_fn for the non-one-hot encoded target case.